lib/src/gotrue_client.dart

```diff
@@ -556,8 +556,11 @@ class GoTrueClient {
         throw AuthException('Session expired.');
       }
     } else {
+      if (_currentSession != null ||
+          _currentSession?.user.id != session.user.id) {
+        _notifyAllSubscribers(AuthChangeEvent.signedIn);
+      }
```

Comment:
This fix is brilliant!

Just one thing, and sorry if I'm missing something, but do we need the `_currentSession != null` here? 

I feel like `_currentSession != null` would be true for every time `recoverSession` is called except for the very first time, and since `_currentSession?.user.id != session.user.id` is true for the very first time, I think we end up notifying subscribers on every `recoverSession` call.

Reply:
Oh good point, I've changed it now to `_currentSession == null`, because if I remember correctly, until the e.g. email is verified you only have a session, but no user, correct?

Reply:
@Vinzent03 
Good catch! Yup, I think that will do it!